### PR TITLE
Revert "make `move` use `=wasMoved` internally (#22032)"

### LINKED ISSUE #22268: `move` now generates an expensive call for every object it's used with
I've been using `move` in my containers as I understand it to be best practise. e.g.

```nim
proc add*[N,T](list: var List[N,T]; item: sink T) =
  ## Add an item to the end of the list.
  boundCheck(list.len < N)
  list.rawPut list.len, move(item)
  inc list.len
```

But, on the latest devel this leads to slow and stack-hungry codegen that won't be inlined.

### Example

```nim
# main.nim

type Foo = object
  a,b,c: int

var dest: Foo

proc main() =
  var x = Foo(a:11, b:12, c:13)
  dest = move(x)

main()
```

### Current Output
Nim generates a `move` implementation and uses it, regardless of whether it's necessary for the object in question.
```c
// system.nim.c
N_LIB_PRIVATE N_NIMCALL(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg, move__main_u8)(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg* x_p0) {
	tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg result;
	nimZeroMem((void*)(&result), sizeof(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg));
	result = (*x_p0);
	eqwasMoved___main_u16(x_p0);
	return result;
}
```
```c
// main.nim.c
N_LIB_PRIVATE N_NIMCALL(void, eqwasMoved___main_u16)(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg* dest_p0) {
	(*dest_p0).a = 0;
	(*dest_p0).b = 0;
	(*dest_p0).c = 0;
}
N_LIB_PRIVATE N_NIMCALL(void, main__main_u6)(void) {
	tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg x;
	nimZeroMem((void*)(&x), sizeof(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg));
	x.a = ((NI)11);
	x.b = ((NI)12);
	x.c = ((NI)13);
	dest__main_u5 = move__main_u8((&x));
}
```

### Expected Output

On [Nim 2.0.0 RC2](https://github.com/nim-lang/nightlies/releases/tag/2023-03-31-version-2-0-2e4ba4ad93c6d9021b6de975cf7ac78e67acba26), codegen is as expected:
```c
// main.nim.c
N_LIB_PRIVATE N_NIMCALL(void, main__main_6)(void) {
	tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg x;
	nimZeroMem((void*)(&x), sizeof(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg));
	x.a = ((NI)11);
	x.b = ((NI)12);
	x.c = ((NI)13);
	dest__main_5 = x;
	nimZeroMem((void*)(&x), sizeof(tyObject_Foo__xRWF2XvRNhnh04uPHdsjXg));
}
```

### Additional Info

Devel:
```
$ nim -v
Nim Compiler Version 1.9.5 [Linux: amd64]
Compiled at 2023-07-06
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 145e002c74c4393b9ac71d5a8cfe066fbf38a048
active boot switches: -d:release
```

RC2:
```
$ nim -v
Nim Compiler Version 1.9.3 [Linux: amd64]
Compiled at 2023-03-31
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 2e4ba4ad93c6d9021b6de975cf7ac78e67acba26
active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by ringabout**
This reverts commit 64b27edd3aa6035bfc5f4e1d83237283bd3b7c2b.

fixes #22268

### CODE REVIEW

**Comment by Araq** [comment]
Hmm, what do you think though? Shouldn't `move` use `wasMoved`? I'm not sure.

**Comment by ringabout** [comment]
I think `move` should use `wasMoved`, but I cannot think of an efficient solution at once, as https://github.com/nim-lang/Nim/issues/22268 pointed out. Perhaps `move` should be a generated hook or something so that it could use the more flexible implementation, namely inline itself when there is no custom `wasMoved` implementation.

File: compiler/ccgexprs.nim [hunk 1]
```diff
@@ -2359,6 +2359,7 @@ proc genMove(p: BProc; n: PNode; d: var TLoc) =
     if d.k == locNone: getTemp(p, n.typ, d)
     if p.config.selectedGC in {gcArc, gcAtomicArc, gcOrc}:
       genAssignment(p, d, a, {})
+      resetLoc(p, a)
```
Comment (Araq) [inline compiler/ccgexprs.nim:2362]: Instead of `resetLoc` we need to call `=wasMoved` here.

Comment (ringabout) [reply compiler/ccgexprs.nim:2362]: Sure thing

Comment (ringabout) [reply compiler/ccgexprs.nim:2362]: Succeeded by https://github.com/nim-lang/Nim/pull/22288

**Pull request closed without merging**